aten/src/ATen/native/im2col_shape_check.h

```diff
@@ -55,7 +54,13 @@ inline void col2im_shape_check(
   int64_t batch_dim = (ndim == 3) ? 0 : -1;
   int64_t n_input_plane = input.size(batch_dim + 1);
 
-  if (n_input_plane % (kernel_width * kernel_height) != 0) {
+  int64_t prod_kernel_size = kernel_height * kernel_width;
```

Comment:
There are `c10::mul_overflows` which could be use to efficiently detect the overflow